homeassistant/components/binary_sensor/rachio.py

```diff
@@ -0,0 +1,117 @@
+"""
+Integration with the Rachio Iro sprinkler system controller.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/binary_sensor.rachio/
+"""
+from abc import abstractmethod
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+from homeassistant.components.rachio import DOMAIN,\
+    EVENT_RACHIO_CONTROLLER_UPDATE,\
+    KEY_DEVICE_ID, KEY_STATUS, KEY_SUBTYPE,\
+    SUBTYPE_OFFLINE, SUBTYPE_ONLINE,\
+    STATUS_OFFLINE, STATUS_ONLINE
+
+DEPENDENCIES = ['rachio']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Rachio binary sensors."""
+    devices = []
+    for controller in hass.data[DOMAIN].controllers:
+        devices.append(RachioControllerOnlineBinarySensor(hass, controller))
+
+    add_devices(devices)
+    _LOGGER.info("%d Rachio binary sensor(s) added", len(devices))
+
+
+class RachioControllerBinarySensor(BinarySensorDevice):
+    """Represent a binary sensor that reflects a Rachio state."""
+
+    def __init__(self, hass, controller, poll=True):
+        """Set up a new Rachio controller binary sensor."""
+        self._controller = controller
+
+        if poll:
+            self._state = self._poll_update()
```

Comment:
This is weird. Why not control this by passing `True` as second parameter to `add_devices` instead. This will cause the `update()` method to be called, which is meant for polling an update

Reply:
With `_poll_update` in the constructor, I can pass data that I already fetched in an earlier call. When HA calls `update`, it would have to fetch that data and use an API call (which now have a rate limit). After instantiation, updates are provided via webhooks only.